test/test.js

```diff
@@ -491,6 +491,39 @@ describe('windows-network-drive', function ()
 					{
 						throw (new Error("Could not get drive list. driveList = " + JSON.stringify(driveList, null, '\t')));
 					}
+
+
+					const driveLetters = Object.keys(driveList)
+					for (let i = 0; i < driveLetters.length; i++) {
+						/**
+						 * The drive letter should be a string of length 1, ie "Z"
+						 */
+						if ( typeof driveLetters[i] !== 'string' || driveLetters[i].length !== 1 ) {
+							throw (new Error("Drive letter is not a string of length 1. driveList = " + JSON.stringify(driveList, null, '\t')));
+						}
+						/**
+						 * The drive letter should be uppercase
+						 */
+						if ( driveLetters[i] !== driveLetters[i].toUpperCase() ) {
+							throw (new Error("Drive letter is not uppercase. driveList = " + JSON.stringify(driveList, null, '\t')));
+						}
+					}
+
+					const firstResult = driveList[driveLetters[0]];
```

Comment:
Why only check the first result? Since you are already looping over the drive letters you could do these checks in the same loop.

Reply:
I am on a small screen so I may have missed it, but I do not see an assertion that forces it to be 1 and only 1. I do see an assertion that will make sure we have at least 1 result.